scripts/build_network_geojson.py

```diff
@@ -0,0 +1,298 @@
+
+    # Assign custom regions if shapefile is provided
+    if region_shapefile != False:
+        regions = gpd.read_file(region_shapefile)
+        bus_gdf = gpd.GeoDataFrame(
+            buses, geometry=[Point(xy) for xy in zip(buses['x'], buses['y'])], crs=regions.crs
+        )
+        bus_gdf = gpd.sjoin(bus_gdf, regions, how="left")
+        buses['region'] = bus_gdf.iloc[:, -1].fillna(buses['country'])  # fallback to country
+    else:
+        buses['region'] = buses['country']
+
+    # Initialize new network
+    agg_network = pypsa.Network()
+
+    # Add one bus per region with coordinates from region_centers
+    for region in buses['region'].unique():
+        if region in region_centers.index:
+            lat, lon = region_centers.loc[region, ['Lat', 'Long']]
+        else:
+            lat, lon = 0.0, 0.0  # fallback if no coords
+        agg_network.add("Bus", name=region, x=lon, y=lat)
+
+    # Aggregate cross-region lines
+    line_data = []
+
+    for _, line in network.lines.iterrows():
+        r0 = buses.loc[line['bus0'], 'region']
+        r1 = buses.loc[line['bus1'], 'region']
+        if r0 != r1:
+            key = tuple(sorted([r0, r1]))
+            line_data.append((key, line['s_nom'], line['length']))
+
+    # Sum s_nom and length per region pair
+    df = pd.DataFrame(line_data, columns=["regions", "s_nom", "length"])
+    grouped = df.groupby("regions").sum().reset_index()
+
+    # Add aggregated lines to network
+    for (r0, r1), row in zip(grouped["regions"], grouped.itertuples()):
+        agg_network.add("Line",
+                        name=f"{r0}_{r1}",
+                        bus0=r0,
+                        bus1=r1,
+                        s_nom=row.s_nom,
+                        length=row.length)
+
+    return agg_network
+
+
+def export_network_lines_to_geojson(network, output_path):
+    """
+    Export the line data of a PyPSA network to a GeoJSON file.
+
+    Parameters
+    ----------
+    network : pypsa.Network
+        The PyPSA network containing line and bus data with coordinates.
+    output_path : str
+        Path to the output GeoJSON file.
+    """
+    features = []
+
+    for name, line in network.lines.iterrows():
+        bus0 = network.buses.loc[line['bus0']]
+        bus1 = network.buses.loc[line['bus1']]
+
+        # Geometry as a LineString between bus coordinates
+        geometry = LineString([(bus0['x'], bus0['y']), (bus1['x'], bus1['y'])])
+
+        # Line properties to include
+        properties = {
+            "name": name,
+            "bus0": line['bus0'],
+            "bus1": line['bus1'],
+            "s_nom": line['s_nom'],
+            "length": line.get('length', None),
+        }
+
+        features.append({"geometry": geometry, "properties": properties})
+
+    gdf = gpd.GeoDataFrame.from_features(features, crs="EPSG:4326")
+    gdf.to_file(output_path, driver="GeoJSON")
+
+
+if __name__ == "__main__":
```

Comment:
What do you think of splitting this script in two scripts (and rules):
1. One to prepare the reference data from electricity statistics
2. The other one to parse the network into a common format

Reply:
Let's keep in mind that it may be useful to have in the output files the clustered pypsa networks similarly to
https://github.com/pypsa-meets-earth/network-comparison/blob/main/notebooks/visual_comparison.ipynb

just saying, not necessary though